core/src/main/java/com/scalar/db/common/CheckedDistributedStorageAdmin.java

```diff
@@ -0,0 +1,277 @@
+package com.scalar.db.common;
+
+import com.scalar.db.api.DistributedStorageAdmin;
+import com.scalar.db.api.TableMetadata;
+import com.scalar.db.exception.storage.ExecutionException;
+import com.scalar.db.io.DataType;
+import com.scalar.db.util.ScalarDbUtils;
+import java.util.Map;
+import java.util.Set;
+import javax.annotation.Nullable;
+
+public class CheckedDistributedStorageAdmin implements DistributedStorageAdmin {
+
+  private final DistributedStorageAdmin admin;
+
+  /**
+   * Whether to check if the namespace exists or not. Set false when the storage does not support
+   * namespaces.
+   */
+  private final boolean checkNamespace;
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin) {
+    this(admin, true);
+  }
+
+  public CheckedDistributedStorageAdmin(DistributedStorageAdmin admin, boolean checkNamespace) {
+    this.admin = admin;
+    this.checkNamespace = checkNamespace;
+  }
+
+  @Override
+  public void createNamespace(String namespace, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace already exists: " + namespace);
+    }
+
+    try {
+      admin.createNamespace(namespace, options);
+    } catch (ExecutionException e) {
+      throw new ExecutionException("Creating the namespace failed: " + namespace, e);
+    }
+  }
+
+  @Override
+  public void createTable(
+      String namespace, String table, TableMetadata metadata, Map<String, String> options)
+      throws ExecutionException {
+    if (checkNamespace && !namespaceExists(namespace)) {
+      throw new IllegalArgumentException("Namespace does not exist: " + namespace);
```

Comment:
When the underlying admin is the DynamoAdmin used with a namespace prefix, the value of the `namespace` argument used in this class here is not prefixed. The error message may be confusing since it will mention the unprefixed namespace. 
For clarity, mentioning the prefixed namespace in the error message throughout this class would be great. What do you think ?

I have the same concern about using the unprefixed namespace in the error message in the ConsensusCommitAdmin

Reply:
I think since users pass the logical name (the unprefixed name) to the APIs, I feel it's consistent and natural for the error messages to use the logical name. In `CheckedDistributedStorageAdmin`, the `IllegalArgumentException` always has the logical name. However, when an error occurs, the message of the exception also uses the logical name, but the cause of the exception should have the physical name.

What do you think?

Reply:
Indeed, this makes sense. Let's keep it as is. Thank you.